adagrams/game.py

```diff
@@ -1,11 +1,179 @@
+#testing commit 
+from copy import deepcopy
+import random
+
+
+
 def draw_letters():
-    pass
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+
+    letter_key_list = list(LETTER_POOL.keys())
+    random.shuffle(letter_key_list)
+    
+    letter_list = []
+    for letter in letter_key_list:
+        if LETTER_POOL[letter] > 0:
+            if len(letter_list) < 10:
+                letter_list.append(letter)
+            
+    return letter_list
+
+#=====another example==========
+# def draw_letters():
+#     # Each time when we cal the function it will create new pool by appending keys v(value)timeseach in our list
+#     # Invoking this function will not change the original pool of letters
+#     #========creating new pool of letters==============
+#     pool=[] 
+#     for k,v in LETTER_POOL.items():
+#         count=0
+#         while count <v:
+#             pool.append(k)
+#             count+=1
 
+#     #======creating hand of letters=================
+#     #hand - random letters that the player has drawn [list of 10]
+#     hand=[]
+#     counter=0
+#     while counter <10:# creating random index from 0 to 10
+#         random_inex=random.randint(0, len(pool)-1)
+#         hand.append(pool[random_inex])   #adding randomly picked letter to our hand (list of 10) 
+#         pool.pop(random_inex)   #removing the same element from the pool 
+#         counter+=1
+#     return hand        
+    
 def uses_available_letters(word, letter_bank):
-    pass
+    pool = deepcopy(letter_bank)
+    pool = [p.upper() for p in pool] # convert each element in a list to uppercase
```

Comment:
Both of these lines are creating a new list.  They can be combined:
```suggestion
    pool = [p.upper() for p in letter_bank] # convert each element in a list to uppercase
```
In addition, a deep copy is not necessary, a shallow copy, ie `letter_bank.copy()` would work here as well to create a copy of the list.  The letters in the letter bank are immutable strings.  A deep copy is only needed when the contents of the list are mutable, and the original data needs to be preserved.